File: boj/18110_solved.ac.js
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const [n, ...input] = require('fs').readFileSync(filePath).toString().trim().split('\n');

function solution(input) {
	if (n === 0) return 0;
	const arr = input.map(Number).sort((a, b) => a - b);
	const splitAverageNum = Math.round(n * 0.15);
	const num = n - splitAverageNum * 2;

	let answer = 0;
	for (let i = splitAverageNum; i < n - splitAverageNum; i += 1) {
		answer += arr[i];
	}

	return Math.round(answer / num);
}

console.log(solution(input));

File: boj/18110_solved.ac.py
import sys


sys.stdin = open("./input.txt", 'r')
n = int(sys.stdin.readline())
input_data = list(map(int, sys.stdin.readlines()))


def round_decimal(num):
    return int(num) + (1 if num - int(num) >= 0.5 else 0)


def solution(input_data):
    if n == 0:
        return 0
    input_data.sort()
    split_average_num = round_decimal(n * 0.15)
    num = n - split_average_num * 2

    total = 0
    for i in range(split_average_num, n - split_average_num):
        total += input_data[i]
    return round_decimal(total / num)


print(solution(input_data))
